bin/ch_core.c

```diff
@@ -68,6 +68,8 @@ struct bind BINDS_DEFAULT[] = {
    { "/var/opt/cray/alps",       "/var/opt/cray/alps",       BD_OPTIONAL },
    /* Cray Shasta/Slingshot bind-mounts. */
    { "/var/spool/slurmd",        "/var/spool/slurmd",        BD_OPTIONAL },
+   /* FLUX job information bind-mount. */
+   { "/var/tmp",                 "/var/tmp",                 BD_OPTIONAL },
```

Comment:
This seems too broad — it makes all of `/var/tmp` shared with the host and you can't turn it off. (Right?) Can we narrow it to just the actual Flux directory?

Reply:
It doesn't appear so. The Flux WLM makes use of a [URI scheme](https://flux-framework.readthedocs.io/projects/flux-core/en/latest/man1/flux-uri.html). The vadar & shared memory related job info are stored as follows.

Note that the hash-looking segments, with exception to the character `ƒ`, are made up. `${USER}` is the user who runs.
```
/var/tmp/${USER}/flux-p5Nca5/jobtmp-0-ƒ56rVV28X/vader_segment.[...]
```

I considered an optional bind-mount of just `/var/tmp/${USER}`, however, unless that user-specific path is created at buildtime, the optional bind wouldn't work and we have the same problem we have now: it won't work with Flux without user manipulation.

Any ideas?

Reply:
Can we bind the Flux directory to something predictable, then tell Flux where it is?

Do the Flux people have any advice?

I don't want to do `/var/tmp` by default as it really is way too broad.

Reply:
> I don't want to do /var/tmp by default as it really is way too broad.

I disagree.

Unlike standard unix desktops or shared environments, HPC resources mount both `/var/tmp` and `/tmp` as tmpfs, and store temporary files only for the duration of the running job. Mounting /var/tmp by default isn't an issue from a HPC job perspective.

I can understand the concern when `/var/tmp` isn't tmpfs, on some shared unix environment, e.g., desktop or front-end, where files from other users are stored via sticky bit and world readable. Charliecloud is for HPC; `ch-run` shouldn't be more difficult to use with Flux because of shared environment concerns. 

FWIW, I think the Flux design decision to use node tmpfs to store node specific job information is very reasonable. Charliecloud should work with FLUX without needing to rely on a FAQ, or tribal knowledge of non-default binds.